### PR TITLE
Restrict CloudSQL instances with public IP

### PR DESCRIPTION
**Pull request by kevensen**
Initial commit of forbiding CloudSQL instances from having a public IP

### CODE REVIEW

**Comment by joecheuk** [comment]
@morgante @t12g  based on our last discussion, it seems to be more natural to fit this to the gcp external ip access constraint template, is that still the case?

https://github.com/forseti-security/policy-library/blob/master/policies/templates/gcp_external_ip_access_v1.yaml

**Comment by morgante** [comment]
@joecheuk I think that constraint should actually be specifically for VMs while we add this new CloudSQL one.

Eventually we could consider a super-constraint for all resource types.

File: validator/sql_public_ip.rego [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package templates.gcp.SQLPUBLICIPV1
```
Comment (morgante) [inline validator/sql_public_ip.rego]: Can you add the Google license header?

Also, package should match constraint name; ie. `GCPSQLPublicIpConstraintV1`.

File: validator/sql_public_ip.rego [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	asset := input.asset
+	asset.asset_type == "sqladmin.googleapis.com/Instance"
+
+	asset.resource.data.settings.ipConfiguration.ipv4Enabled == true
```
Comment (morgante) [inline validator/sql_public_ip.rego]: Is the `ipConfiguration` object always present?

File: validator/test/fixtures/constraints/sql_public_ip/data.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: constraints.gatekeeper.sh/v1alpha1
```
Comment (morgante) [inline validator/test/fixtures/constraints/sql_public_ip/data.yaml]: Needs license.

File: validator/sql_public_ip.rego (hunk 2 above)
Comment (kevensen) [reply validator/sql_public_ip.rego]: Yes

File: policies/templates/gcp_sql_public_ip_v1.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        kind: GCPSQLPublicIpConstraintV1
+        plural: gcpsqlpublicipconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties: {}
+  targets:
+   validation.gcp.forsetisecurity.org:
+      rego: |
+            #INLINE("validator/sql_public_ip.rego")
+           package templates.gcp.SQLPUBLICIPV1
```
Comment (morgante) [inline policies/templates/gcp_sql_public_ip_v1.yaml]: Please run `make build` again.

**Pull request merged**